browser/src/control/Control.FormulaUsagePopup.ts

```diff
@@ -43,10 +43,16 @@ class FormulaUsagePopup extends L.Control.AutoCompletePopup {
 	onAdd() {
 		this.newPopupData.isAutoCompletePopup = true;
 		this.map.on('openformulausagepopup', this.openFormulaUsagePopup, this);
-		this.map.on('closeformulausagepopup', this.closePopup, this);
+		this.map.on('closeformulausagepopup', this.closeFormulaUsagePopup, this);
 		this.map.on('sendformulausagetext', this.sendFormulaUsageText, this);
 	}
 
+	closeFormulaUsagePopup() {
+		var formulaUsagePopup = L.DomUtil.get('formulausagePopup');
```

Comment:
can be that if condition doen inside super.closePopup ?
possibly we already have that id in the base class?

Reply:
then we will not need also previous similar code in the autocomplete popup too

Reply:
But closePopup is a base class for AutoFill Mention apart from function autocomplete and usage, If i put a condition into base class it will look like following
if (id==formulausagePopup || formulaautocompletePopup)
    do the close action
But autofill and mention will fail this case. I have to add all ids. Does it make sense?
Actually it will work without any condition but it can trig many unnecessary operation.
I think this is best version of code. What do you think?

Reply:
I mean we can store "id" as a member in the base class when we create it.
Then base class will know the id and we will not have to add "if"s.
We will just do something like: `L.DomUtil.get(this.baseId)`

Reply:
and subclasses can adjust the functionality:
```
closePopup() {
    super.closePopup();
    doAdditionalStuffInThisSubclass();
}
```